System/Random.hs

```diff
@@ -20,41 +20,181 @@
 -- Module      :  System.Random
 -- Copyright   :  (c) The University of Glasgow 2001
 -- License     :  BSD-style (see the file LICENSE in the 'random' repository)
---
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  stable
--- Portability :  portable
 --
--- This library deals with the common task of pseudo-random number
--- generation. The library makes it possible to generate repeatable
--- results, by starting with a specified initial random number generator,
--- or to get different results on each run by using the system-initialised
--- generator or by supplying a seed from some other source.
+-- This library deals with the common task of pseudo-random number generation.
+--
+-- = Overview
+-- This library provides typeclasses and instances for the following concepts:
+--
+-- [Pure pseudo-random number generators] 'RandomGen' is an interface to pure
+--     pseudo-random number generators.
```

Comment:
You should mention here that these are splitable pseudo-random number generator.

Reply:
I believe the way we're thinking about them now is that they are _potentially_ splittable pure pseudo-random number generators. That is, if the PRNG algorithm has no "split" method, then `split` should not be implemented, or alternatively it should throw.

Actually saying "potentially splittable pure pseudo-random number generators" seems like a bit of a mouthful here, so I haven't changed the text here. I have added something about splittable PRNGs in the section "Pure and monadic pseudo-random number generators".